feat(source-pg): New Postgres source plugin with CDC (#7975)
cloudquery/cloudquery: website/pages/docs/plugins/sources/postgresql/overview.mdx

---
name: Postgres
---
# Postgres Source Plugin

import { getLatestVersion } from "../../../../../utils/versions";
import { Badge } from "../../../../../components/Badge";
import { Callout } from 'nextra-theme-docs'

<Badge text={"Latest: " + getLatestVersion("source", "postgresql")}/>

The CloudQuery Postgres plugin syncs your Postgres database to any of the supported CloudQuery destinations (e.g. PostgreSQL, Snowflake, BigQuery, …).
This plugins also supports CDC via Postgres logical replication which enables keeping your Postgres up to date with any destination just by subscribing to changes.

Supported database versions:

> - PostgreSQL >= v11

Author: Yes

---
name: PostgreSQL
---
# PostgreSQL Source Plugin

import { getLatestVersion } from "../../../../../utils/versions";
import { Badge } from "../../../../../components/Badge";
import { Callout } from 'nextra-theme-docs'

<Badge text={"Latest: " + getLatestVersion("source", "postgresql")}/>

> The CloudQuery PostgreSQL plugin syncs your PostgreSQL database to any of the supported CloudQuery destinations (e.g. PostgreSQL, Snowflake, BigQuery, …).

Reviewer: ```suggestion
The CloudQuery PostgreSQL plugin syncs your PostgreSQL database to any of the supported CloudQuery destinations (e.g. PostgreSQL, BigQuery, Snowflake, and [more](https://www.cloudquery.io/docs/plugins/destinations/overview)).
```

> This plugins also supports CDC via PostgreSQL logical replication which enables keeping your PostgreSQL up to date with any destination just by subscribing to changes.

Reviewer: ```suggestion
This plugin also supports CDC via PostgreSQL logical replication, which enables keeping your PostgreSQL up to date with any destination by subscribing to changes.
```

  - `"postgres://jack:secret@localhost:5432/mydb?sslmode=prefer"` _connect with tcp and prefer TLS_
  - `"postgres://jack:secret@localhost:5432/mydb?sslmode=disable&application_name=pgxtest&search_path=myschema&connect_timeout=5"` _be explicit with all options_
  - `"postgres://localhost:5432/mydb?sslmode=disable"` _connect with os username cloudquery is being run as_
  - `"postgres:///mydb?host=/tmp"` _connect over unix socket_
  - `"dbname=mydb"` _unix domain socket, just specifying the db name - useful if you want to use peer authentication_
  - `"user=jack password=jack\\'ssooper\\\\secret host=localhost port=5432 dbname=mydb sslmode=disable"` _DSN with escaped backslash and single quote_

- `pgx_log_level` (string, optional. Default: "error")

  Available: "error", "warn", "info", "debug", "trace"
  define if and in which level to log [`pgx`](https://github.com/jackc/pgx) call.

- `cdc` (bool, optional. Default: false)

  If enabled the source plugin will start syncing CDC via PostgreSQL logical replication in real-time.

> Note: Make sure you use environment variable expansion in production instead of committing the credentials to the configuration file directly.

Reviewer: ```suggestion
```

This is already mentioned in a callout higher up

### Verbose logging for debug

The PostgreSQL source can be run in debug mode.

> Note: This will use [`pgx`](https://github.com/jackc/pgx) built-in logging and might output data/sensitive information to logs so make sure to not use it in production but only for debugging.

Reviewer: ```suggestion
Note: This will use [`pgx`](https://github.com/jackc/pgx) built-in logging and might output data/sensitive information to logs so make sure to not use it in production, only for debugging.
```
